packages/atomic/src/components/insight/atomic-insight-result/atomic-insight-result.tsx

```diff
@@ -123,40 +123,6 @@ export class AtomicInsightResult {
       .join('');
   }
 
```

Comment:
This is a good refacto 🙏 , magic ✨ 
but one only thing, deleting this selectors make the css global and not only when we have result actions implemented and on hover. for example the hover on each result changes color all the time even if we have no result action to show. I personnaly don't find it disturbing to be global, what do you think ? 

case on hover with no result action:

![Screenshot 2023-07-07 at 11 40 11 AM](https://github.com/coveo/ui-kit/assets/119955059/34015e7c-8a9b-43f7-ad2e-960304f8e4d0)

Reply:
Good catch, missed it.
Yeah, I think having global is fine too. Maybe validate with the rest of the gang?

If that's a no-go, right now there's no way to do it in CSS only (we need the `:has` pseudo-selector for that, and Firefox does not support it yet). I would then put a CSS class on the host element "hasResultAction" and a TODO to refactor when Firefox supports it).

https://developer.mozilla.org/en-US/docs/Web/CSS/:has